### PR TITLE
msgpack-c: switch build system to cmake

### PR DESCRIPTION
**Pull request by ys-nuem**
From next release, Msgpack is going to stop supporting for autotools (see https://github.com/msgpack/msgpack-c/issues/469).
The purpose of this PR is to make updating to the next release smoothly.

It also includes minor updates to 1.4.2.

### CODE REVIEW

File: mingw-w64-msgpack-c/PKGBUILD [hunk 1]
```diff
@@ -23,18 +25,31 @@ prepare() {
 
 build() {
   cd "msgpack-${pkgver}"
-  ./configure --prefix=${MINGW_PREFIX}
+
+  MSYS2_ARG_CONV_EXCL='-DCMAKE_INSTALL_PREFIX=' \
+    ${MINGW_PREFIX}/bin/cmake \
+      -G "Unix Makefiles" \
```
Comment (Alexpux) [inline mingw-w64-msgpack-c/PKGBUILD]: Here need to be "MSYS Makefiles"

**Comment by ys-nuem** [comment]
Additional changes are:
* fix generator to "MSYS Makefiles"
* change the temporary path of build directory

File: mingw-w64-msgpack-c/PKGBUILD [hunk 2]
```diff
@@ -1,40 +1,55 @@
...
 }
 
 package() {
-  cd "msgpack-${pkgver}"
-  make DESTDIR="$pkgdir" install
+  cd "${srcdir}/build-${MINGW_CHOST}"
+  make DESTDIR="${pkgdir}" install
+
+  mkdir -p "${pkgdir}${MINGW_PREFIX}"/bin
+  mv "${pkgdir}${MINGW_PREFIX}"/lib/libmsgpackc.dll "${pkgdir}${MINGW_PREFIX}"/bin
```
Comment (Alexpux) [inline mingw-w64-msgpack-c/PKGBUILD]: this is need to be fixed in cmake scripts instead manually doing this

File: mingw-w64-msgpack-c/PKGBUILD [hunk 3]
```diff
@@ -1,40 +1,55 @@
...
+
+  mv "${pkgdir}${MINGW_PREFIX}"/lib/{libmsgpackc_import.lib,libmsgpackc.dll.a}
```
Comment (Alexpux) [inline mingw-w64-msgpack-c/PKGBUILD]: This is too can be fixed in cmake scripts

**Comment by ys-nuem** [comment]
All scripts in `package()` have changed to paches of CMakeFiles.txt

**Pull request merged**